beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3068,6 +3060,10 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
 
         drop(lag_timer);
 
+        // Find the dependent roots associated with this head before updating the snapshot. This
+        // is to ensure consistency when sending server sent events later in this method.
+        let dependent_roots = new_head.beacon_state.get_dependent_roots();
```

Comment:
Maybe we could bump this up inside the `lag_timer`. I doubt it's going to take any significant time, but it's consistent.